### ISSUE TITLE
stage2 usingnamespace with private decl causes ambiguous reference

### ISSUE THREAD

**Issue by billzez** [ziglang/zig#13062]
### Zig Version

0.10.0-dev.4217+9d8cdb855

### Steps to Reproduce

```
const std = @import("std");

pub const Mixin = struct {
    const X = 1;
};

pub const Thing = struct {
    pub usingnamespace Mixin;
    pub const X = 1;
};

pub fn main() void {
    std.debug.print("{d}\n", .{Thing.X});
}
```

### Expected Behavior

Compiles

### Actual Behavior

```
$ zig build-exe main.zig 
main.zig:13:37: error: ambiguous reference
    std.debug.print("{d}\n", .{Thing.X});
                               ~~~~~^~
main.zig:9:9: note: declared here
    pub const X = 1;
    ~~~~^~~~~~~~~~~
main.zig:4:5: note: declared here
    const X = 1;
    ^~~~~~~~~~~
```

**Comment by nektro** [comment]
working as intended imo

**Comment by billzez** [comment]
Desired or not, it's surprising that `usingnamespace` consumes private declarations.  It's a change in behavior from stage1 and not documented.

**Comment by nektro** [comment]
oh no, you're right. that is a bug. `Thing.X` should unambiguously reference only the `pub` one

**Comment by Vexu** [comment]
`pub` only matters if you put `Mixin` in a separate file in which case the reference becomes unambiguous and compiles. If everything is in the same file then the compile error is correct.

**Comment by billzez** [comment]
@Vexu - that's actually how I hit this bug.  It was in a separate file, and I was confused how a private const in one file used as a mixin was affected by the file that consumed it.

**Comment by billzez** [comment]
Yes, #13077 looks like my original issue.  I though the above was the simplified version, but it appears this issue and that are distinct.  I'm not advocating that this above issue be fixed, but I do find it confusing.

**Comment by McSinyx** [comment]
This can probably be closed now.
